.github/workflows/pypi-publish.yml

```diff
@@ -25,11 +25,23 @@ jobs:
     permissions:
       id-token: "write"
     steps:
+      - name: Get publish-requirements.txt from repository
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+        with:
+          sparse-checkout: |
+            .github/workflows/requirements/publish-requirements.txt
+          sparse-checkout-cone-mode: false
+          # The tag to build or the tag received by the tag event
```

Comment:
Is this comment actually correct? I think the only time this will be anything besides main is a workflow_dispatch with manual branch setting. I believe on workflow_run (which is how it'll happen on a real release), it'll always be main.